tests/fsharp/Compiler/CompilerAssert.fs

```diff
@@ -52,12 +56,18 @@ type CompileOutput =
     | Library
     | Exe
 
-type CompilationReference = private CompilationReference of Compilation * staticLink: bool with
+type CompilationReference = 
+    private 
```

Comment:
nit: `type private` is more idiomatic

Reply:
`type private` makes the whole type `private` which is not what I want. I only want to make the union cases private to the module.